<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Library Management System</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QMainWindow {
    background-color: #f5f5f5;
}

QTableWidget {
    background-color: white;
    alternate-background-color: #f0f0f0;
    border: 1px solid #d0d0d0;
    border-radius: 5px;
    selection-background-color: #3498db;
    selection-color: white;
	color: black;
}

QTableWidget::item {
    padding: 5px;
}

QPushButton {
    background-color: #3498db;
    color: white;
    border: none;
    border-radius: 5px;
    padding: 8px 16px;
    font-weight: bold;
}

QPushButton:hover {
    background-color: #2980b9;
}

QPushButton:pressed {
    background-color: #1c6ea4;
}

QPushButton:disabled {
    background-color: #bdc3c7;
}

QHeaderView::section {
    background-color: #3498db;
    color: white;
    padding: 5px;
    border: none;
}

QLabel {
    font-size: 14px;
    color: #2c3e50;
}

QGroupBox {
    font-weight: bold;
    border: 1px solid #d0d0d0;
    border-radius: 5px;
    margin-top: 10px;
    padding-top: 15px;
}

QGroupBox::title {
    subcontrol-origin: margin;
    subcontrol-position: top center;
    padding: 0 5px;
}

QStatusBar{
	background-color:white;
}
QMenuBar{
	background-color: #3498db;
}

QLineEdit{
	color:black;
}

QMessabeBox{
	color:black;
	background-color:white;
}</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="styleSheet">
    <string notr="true">QWidget{
	background-color: white;
}</string>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QLabel" name="labelTitle">
      <property name="text">
       <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;&lt;span style=&quot; font-size:16pt; font-weight:600;&quot;&gt;Library Management System&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignmentFlag::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QTableWidget" name="booksTableWidget">
      <property name="alternatingRowColors">
       <bool>true</bool>
      </property>
      <property name="selectionMode">
       <enum>QAbstractItemView::SelectionMode::SingleSelection</enum>
      </property>
      <property name="selectionBehavior">
       <enum>QAbstractItemView::SelectionBehavior::SelectRows</enum>
      </property>
      <column>
       <property name="text">
        <string>IDss</string>
       </property>
       <property name="textAlignment">
        <set>AlignJustify|AlignBottom</set>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Title</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Author</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Genre</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Publication Year</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Status</string>
       </property>
      </column>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBoxSearch">
      <property name="styleSheet">
       <string notr="true">QGroupBox{
	color: black;
}

QComboBox{
	background-color: #3498db;
	color:black;
}</string>
      </property>
      <property name="title">
       <string>Search Books</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="QLabel" name="labelSearch">
         <property name="text">
          <string>Search:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="lineEditSearch">
         <property name="placeholderText">
          <string>Enter title, author or keyword...</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="comboBoxSearchType">
         <property name="styleSheet">
          <string notr="true">QComboBox {
    background-color: #3498db;
    color: black;
}

QComboBox QAbstractItemView {
    background-color: #3498db; 
    color: black;              
    selection-background-color: #2980b9;
    selection-color: white;      
}
</string>
         </property>
         <item>
          <property name="text">
           <string>Title</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Author</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Genre</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>All Fields</string>
          </property>
         </item>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="ResultLabel">
         <property name="styleSheet">
          <string notr="true">QLabel{
	color:red;
}</string>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBoxActions">
      <property name="styleSheet">
       <string notr="true">QGroupBox{
	color: black;
}

QPushButton {
    background-color: #3498db;
    color: white;
    border: none;
    border-radius: 5px;
    padding: 8px 16px;
    font-weight: bold;
}

QPushButton:hover {
    background-color: #2980b9;
}

QPushButton:pressed {
    background-color: #1c6ea4;
}

QPushButton:disabled {
    background-color: #bdc3c7;
}</string>
      </property>
      <property name="title">
       <string>Book Actions</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>15</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QLabel" name="labelUserId">
         <property name="text">
          <string>Selected Book</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="SelectedLabel">
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="borrowButton">
         <property name="minimumSize">
          <size>
           <width>140</width>
           <height>0</height>
          </size>
         </property>
         <property name="text">
          <string>Borrow Book</string>
         </property>
         <property name="icon">
          <iconset theme="go-down"/>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="returnButton">
         <property name="minimumSize">
          <size>
           <width>140</width>
           <height>0</height>
          </size>
         </property>
         <property name="text">
          <string>Return Book</string>
         </property>
         <property name="icon">
          <iconset theme="go-up"/>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="refreshButton">
         <property name="minimumSize">
          <size>
           <width>140</width>
           <height>0</height>
          </size>
         </property>
         <property name="text">
          <string>Refresh List</string>
         </property>
         <property name="icon">
          <iconset theme="view-refresh"/>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_2">
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>15</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>24</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionRefresh"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuBooks">
    <property name="title">
     <string>Books</string>
    </property>
    <addaction name="actionAddBook"/>
    <addaction name="actionEditBook"/>
    <addaction name="actionDeleteBook"/>
    <addaction name="separator"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuBooks"/>
   <addaction name="menuHelp"/>
  </widget>
  <action name="actionAddBook">
   <property name="text">
    <string>Add Book</string>
   </property>
  </action>
  <action name="actionEditBook">
   <property name="text">
    <string>Edit Book</string>
   </property>
  </action>
  <action name="actionDeleteBook">
   <property name="text">
    <string>Delete Book</string>
   </property>
  </action>
  <action name="actionBorrowBook">
   <property name="text">
    <string>Borrow Book</string>
   </property>
  </action>
  <action name="actionReturnBook">
   <property name="text">
    <string>Return Book</string>
   </property>
  </action>
  <action name="actionRefresh">
   <property name="text">
    <string>Refresh</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
